### PR TITLE
106872 change category endpoint

### PR DESCRIPTION
**Pull request by janid1967**
[AB#106872](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/106872)

### CODE REVIEW

File: src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/UpdatePunchItemCategory/UpdatePunchItemCategoryCommandHandler.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    {
+        _punchItemRepository = punchItemRepository;
+        _unitOfWork = unitOfWork;
+        _logger = logger;
+    }
+
+    public async Task<Result<string>> Handle(UpdatePunchItemCategoryCommand request, CancellationToken cancellationToken)
+    {
+        var punchItem = await _punchItemRepository.GetByGuidAsync(request.PunchItemGuid);
+        if (punchItem is null)
```
Comment (jsoi-eq) [inline src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/UpdatePunchItemCategory/UpdatePunchItemCategoryCommandHandler.cs:34]: Do we need this check if UpdatePunchItemCategoryCommandValidator is always run before the Handle method?

File: src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/UpdatePunchItemCategory/UpdatePunchItemCategoryCommandValidator.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        async Task<bool> NotBeInAVoidedTagForPunchItemAsync(Guid punchItemGuid, CancellationToken cancellationToken)
+            => !await punchItemValidator.TagOwningPunchItemIsVoidedAsync(punchItemGuid, cancellationToken);
+
+        async Task<bool> BeAnExistingPunchItemAsync(Guid punchItemGuid, CancellationToken cancellationToken)
+            => await punchItemValidator.ExistsAsync(punchItemGuid, cancellationToken);
+
+        async Task<bool> NotAlreadyBeClearedAsync(Guid punchItemGuid, CancellationToken cancellationToken)
+            => !await punchItemValidator.IsClearedAsync(punchItemGuid, cancellationToken);
+
+        async Task<bool> NotHaveCategoryAsync(Guid punchItemGuid, Category category, CancellationToken cancellationToken)
```
Comment (jsoi-eq) [inline src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/UpdatePunchItemCategory/UpdatePunchItemCategoryCommandValidator.cs:43]: Maybe NotHaveSameCategoryAsync?

I was thrown multiple times here and in unit tests by NotHaveCategory, interpreting it unconsciously  as requiring punchItem to have an empty/null Category property, although if I consciously inspect the class property I see it is non-nullable.
Update unit test names also if you choose to update this name.

**Review by jsoi-eq — COMMENTED** [review]
Awaiting feedback.

File: src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/UpdatePunchItemCategory/UpdatePunchItemCategoryCommandHandler.cs (hunk 1 above)
Comment (janid1967) [reply src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/UpdatePunchItemCategory/UpdatePunchItemCategoryCommandHandler.cs:34]: Agree. Removed

**Review by jsoi-eq — APPROVED** [review]

**Pull request merged**